src/Workspaces/Core/Portable/Rename/RenameUtilities.cs

```diff
@@ -183,6 +183,25 @@ private static bool ShouldRenameOnlyAffectDeclaringProject(ISymbol symbol)
             return TokenRenameInfo.NoSymbolsTokenInfo;
         }
 
+        /// We try to rewrite all locations that are invalid candidate locations. If there is only
+        /// one location it must be the correct one (the symbol is ambiguous to something else)
+        /// and we always try to rewrite it.  If there are multiple locations, we only allow it
+        /// if the candidate reason allows for it).
+        internal static bool ShouldIncludeLocation(ImmutableArray<RenameLocation> renameLocations, RenameLocation location)
+        {
+            if (location.IsRenameInStringOrComment)
+            {
+                return false;
+            }
+
+            if (renameLocations.Length == 1)
```

Comment:
Why do we always allow if the locations array is size 1? Shouldn't it still need to go through `RenameLocation.ShouldRename`?

Reply:
It's copy-paste code : ) So I don't know the original reason.

Reply:
Trying to read the commend alone, I'm guessing this is some sort of handling around doing a rename in broken code scenarios. If you're doing a rename of a method and you hit a location where overload resolution as broken before and there were multiple candidates, it's ambiguous if you should rename that location or not. But if overload resolution only said there was one candidate, you just assume that's what the user meant.